internal/engine/buildcontrol/build_control.go

```diff
@@ -39,6 +39,21 @@ func NextTargetToBuild(state store.EngineState) (*store.ManifestTarget, HoldSet)
 		}
 	}
 
+	// We do not know whether targets are enabled or disabled until their configmaps + uiresources are synced
+	// and reconciled. This happens very quickly after the first Tiltfile execution.
+	// If any targets have an unknown EnableStatus, then we don't have enough information to schedule builds:
+	// - If we treat an unknown as disabled but it is actually enabled, then we break our heuristic prioritization
+	//   (e.g., we might schedule k8s resources before local resources).
+	// - If we treat an unknown as enabled but it is actually disabled, then we start logging + side-effecting
+	//   a build that might immediately be canceled.
+	if pending := TargetsWithPendingEnableStatus(targets); len(pending) > 0 {
+		holds.Fill(targets, store.Hold{
+			Reason: store.HoldReasonResourceInitializing,
```

Comment:
I'm not in love with this name and open to alternatives.
Another option is to not even hold, just return nil - is it better to briefly show a potentially confusing hold reason, or just show nothing and risk silently holding forever? Maybe hiding this reason would be better after all. If things work as expected, it should only ever flash for an unreadably short amount of time.

Reply:
i'd probably use `HoldReasonTiltfileReload`...that's a good user-level description of what's actually going on (we're still processing the tiltfile execution)...we don't need to have detailed steps of how the tiltfile execution registers objects with the apiserver.